### ISSUE TITLE
Use of qualified function names in @linq

### ISSUE THREAD

**Issue by bkamins** [JuliaStats/DataFramesMeta.jl#94]
Following https://stackoverflow.com/questions/50547688/julia-using-groupby-with-dataframesmeta-and-lazy.

Would it be possible to add support for passing qualified function names to `@linq` macro?

**Comment by tshort** [comment]
If I'm reading it right, `@linq` works. It's use of Lazy that has the conflict. The main options I see are:

1. Remove all uses of Lazy from the docs/examples, and encourage `@linq`.

2. Pull in `@>` from Lazy.

**Comment by bkamins** [comment]
Yes - I do not say that `@linq` is broken. I am making a feature request 😄. Incidentally this request is needed with Lazy.jl because of `groupby` name clash.

Let me explain the problem more explicitly. Assume you do not want to pollute a namespace for instance (which can be expected in production code) and use `import` instead of `using`. How would you use `@linq` then?

```
julia> import DataFrames, DataFramesMeta

julia> x = DataFrames.DataFrame(rand(3,4))
3×4 DataFrames.DataFrame
│ Row │ x1       │ x2       │ x3       │ x4       │
├─────┼──────────┼──────────┼──────────┼──────────┤
│ 1   │ 0.789172 │ 0.2509   │ 0.107305 │ 0.504798 │
│ 2   │ 0.173546 │ 0.847316 │ 0.292337 │ 0.880935 │
│ 3   │ 0.304413 │ 0.460651 │ 0.368497 │ 0.581835 │

julia> DataFramesMeta.@linq x |> DataFrames.groupby(:x1)
ERROR: MethodError: Cannot `convert` an object of type Expr to an object of type DataFramesMeta.SymbolParameter
This may have arisen from a call to the constructor DataFramesMeta.SymbolParameter(...),
since type constructors fall back to convert methods.
```

whereas with `using` you have:

```
julia> using DataFrames, DataFramesMeta

julia> @linq x |> groupby(:x1)
DataFrames.GroupedDataFrame  3 groups with keys: Symbol[:x1]
First Group:
1×4 DataFrames.SubDataFrame{Array{Int64,1}}
│ Row │ x1       │ x2     │ x3       │ x4       │
├─────┼──────────┼────────┼──────────┼──────────┤
│ 1   │ 0.789172 │ 0.2509 │ 0.107305 │ 0.504798 │
⋮
Last Group:
1×4 DataFrames.SubDataFrame{Array{Int64,1}}
│ Row │ x1       │ x2       │ x3       │ x4       │
├─────┼──────────┼──────────┼──────────┼──────────┤
│ 1   │ 0.304413 │ 0.460651 │ 0.368497 │ 0.581835 │
```

So in short a feature request is for `@linq` to properly resolve qualified function names in general (and `groupby` is only a special case - coincidentally referred to in the documentation).

Of course - you might get name clash of any function you use in general (there is no guarantee that some other package does not use the same name for something else) - so I would find this general qualified name resolution functionality desirable.

**Comment by tshort** [comment]
Got it. That is a problem with `@linq`. If you care to have a go at a fix, the following link shows where the appropriate code is. That needs to be generalized to handle qualified functions.

https://github.com/JuliaStats/DataFramesMeta.jl/blob/b016059cce76773ca0382dbaed15eb13e1d71f2d/src/linqmacro.jl#L70-L84

**Comment by bkamins** [comment]
In #95 I have proposed the fix.
